packages/react/src/Tabs/TabsButton.test.tsx

```diff
@@ -52,6 +53,26 @@ describe('Tabs button', () => {
     expect(component).toHaveAttribute('aria-controls', '-panel-0')
   })
 
+  it('calls onChange with the newly activated tab', () => {
+    const onChange = jest.fn()
+    const contextValue = {
+      activeTab: 0,
+      tabsId: 'test-tabs',
+      updateTab: onChange,
+    }
+
+    render(
+      <TabsContext.Provider value={contextValue}>
+        <TabsButton tab={1}>Tab 1</TabsButton>
+      </TabsContext.Provider>,
+    )
+
+    const button = screen.getByRole('tab', { name: 'Tab 1' })
+    fireEvent.click(button)
```

Comment:
It's recommended to use [user-event](https://testing-library.com/docs/user-event/intro/) to simulate user events